temporal-testing-junit5/src/main/java/io/temporal/testing/TestWorkflowExtension.java

```diff
@@ -0,0 +1,323 @@
+    return context.getStore(namespace);
+  }
+
+  public static class Builder {
+
+    private static final Class<?>[] NO_WORKFLOWS = new Class<?>[0];
+    private static final Object[] NO_ACTIVITIES = new Object[0];
+
+    private WorkerOptions workerOptions = WorkerOptions.getDefaultInstance();
+    private String namespace = "UnitTest";
+    private Class<?>[] workflowTypes = NO_WORKFLOWS;
+    private Object[] activityImplementations = NO_ACTIVITIES;
+    private boolean useExternalService = false;
+    private String target = null;
+    private boolean doNotStart = false;
+
+    private Builder() {}
+
+    /** @see TestWorkflowEnvironment#newWorker(String, WorkerOptions) */
+    public Builder setWorkerOptions(WorkerOptions options) {
+      this.workerOptions = options;
+      return this;
+    }
+
+    /**
+     * Set Temporal namespace to use for tests, by default, {@code UnitTest} is used.
+     *
+     * @see WorkflowClientOptions#getNamespace()
+     */
+    public Builder setNamespace(String namespace) {
+      this.namespace = namespace;
+      return this;
+    }
+
+    /**
+     * Specify workflow implementation types to register with the Temporal worker.
+     *
+     * @see Worker#registerWorkflowImplementationTypes(Class[])
+     */
+    public Builder setWorkflowTypes(Class<?>... workflowTypes) {
+      this.workflowTypes = workflowTypes;
+      return this;
+    }
+
+    /**
+     * Specify activity implementations to register with the Temporal worker
+     *
+     * @see Worker#registerActivitiesImplementations(Object...)
+     */
+    public Builder setActivityImplementations(Object... activityImplementations) {
+      this.activityImplementations = activityImplementations;
+      return this;
+    }
+
+    /**
+     * Switches to external Temporal service implementation with default endpoint of {@code
+     * 127.0.0.1:7233}.
+     *
+     * @see TestEnvironmentOptions.Builder#setUseExternalService(boolean)
+     * @see TestEnvironmentOptions.Builder#setTarget(String)
+     * @see WorkflowServiceStubsOptions.Builder#setTarget(String)
+     */
+    public Builder useExternalService() {
+      return useExternalService(null);
+    }
+
+    /**
+     * Switches to external Temporal service implementation.
+     *
+     * @param target defines the endpoint which will be used for the communication with standalone
+     *     Temporal service.
+     * @see TestEnvironmentOptions.Builder#setUseExternalService(boolean)
+     * @see TestEnvironmentOptions.Builder#setTarget(String)
+     * @see WorkflowServiceStubsOptions.Builder#setTarget(String)
+     */
+    public Builder useExternalService(String target) {
+      this.useExternalService = true;
+      this.target = target;
+      return this;
+    }
+
+    /** Switches to internal in-memory Temporal service implementation (default). */
+    public Builder useInternalService() {
+      this.useExternalService = false;
+      this.target = null;
+      return this;
+    }
```

Comment:
I've changed the way `target` and `useExternalService` are to make combinations like `useExternalService = false`, `target != null` inexpressible via the builder. Should this schema be kept or reverted to the usual builder format?